joplin/base/models/janis_page.py

```diff
@@ -106,6 +108,21 @@ def janis_url(self):
 
             # add hardcoded language path to base url
             base_url = os.environ["JANIS_URL"] + '/en'
+            # attributes for the url are needed by not discovered yet lets fetch them
+            # looking for missing elements, deducing content type from what works and what dosen't
+            # this is pretty ugly and ought to be cleaned up
+            if theme_slug is None and self.content_type.name != 'department page':
+                try:
+                    theme_slug = self.topics.first().topic.topiccollections.first().topiccollection.theme.slug or None
+                    topic_collection_slug = self.topics.first().topic.topiccollections.first().topiccollection.slug or None
+                except AttributeError as e:
+                    try:
+                        theme_slug = self.topiccollections.first().topiccollection.theme.slug or None
+                        topic_collection_slug = self.topiccollections.first().topiccollection.slug or None
+                    except AttributeError as e:
+                        # this is for pages just under departments
+                        theme_slug = self.related_departments.all()[0].related_department.slug or None
```

Comment:
not the biggest fan of having department named theme here, shouldn't the entire URL be /department_slug/page_slug in this scenario?

Reply:
Thats the effective result. I agree that as is isn’t super legible, but abstractly ‘theme_slug’ is just the first path